src/core/coap/coap.cpp

```diff
@@ -672,20 +692,32 @@ void Coap::ProcessReceivedRequest(Header &aHeader, Message &aMessage, const Ip6:
         if (strcmp(resource->mUriPath, uriPath) == 0)
         {
             resource->HandleRequest(aHeader, aMessage, aMessageInfo);
+            error = kThreadError_None;
             ExitNow();
         }
     }
 
     if (mDefaultHandler)
     {
         mDefaultHandler(mDefaultHandlerContext, &aHeader, &aMessage, &aMessageInfo);
+        error = kThreadError_None;
```

Comment:
Yes, I see a couple of options here:

1. Update the default handler to allow returning an error - this would allow the default handler to return `kThreadError_NotFound` and utilize the same "Not Found" error response code path here, or
2. Assume that the application will take on full responsibility in generating all responses, which is what is done here and what you've suggested above.

I'm good with option 2 in that it limits the scope of OpenThread.